test/fhirtypes/ElementDefinition.fixInstanceDefinition.test.ts

```diff
@@ -0,0 +1,169 @@
+      );
+      expect(valueX.patternPeriod).toBeUndefined();
+      expect(valueX.fixedPeriod).toBeUndefined();
+    });
+
+    it('should throw ValueAlreadyFixedError when the value is fixed to a different value by pattern[x]', () => {
+      const workAddress = new InstanceDefinition();
+      workAddress._instanceMeta.name = 'USPostalAddress';
+      workAddress._instanceMeta.sdType = 'Address';
+      workAddress._instanceMeta.usage = 'Inline';
+      workAddress.country = 'US';
+      workAddress.type = 'postal';
+      workAddress.use = 'work';
+      const homeAddress = new InstanceDefinition();
+      homeAddress._instanceMeta.name = 'USPostalAddress';
+      homeAddress._instanceMeta.sdType = 'Address';
+      homeAddress._instanceMeta.usage = 'Inline';
+      homeAddress.country = 'US';
+      homeAddress.type = 'postal';
+      homeAddress.use = 'home';
+
+      const address = patient.elements.find(e => e.id === 'Patient.address');
+      address.fixValue(workAddress);
+      expect(address.patternAddress).toEqual({ country: 'US', type: 'postal', use: 'work' });
+      address.fixValue(workAddress);
+      expect(address.patternAddress).toEqual({ country: 'US', type: 'postal', use: 'work' });
+      expect(() => {
+        address.fixValue(homeAddress);
+      }).toThrow(
+        'Cannot fix {"country":"US","type":"postal","use":"home"} to this element; a different Address is already fixed: {"country":"US","type":"postal","use":"work"}'
+      );
+    });
+
+    it('should throw ValueAlreadyFixedError when the value is fixed to a different value by fixed[x]', () => {
+      const workAddress = new InstanceDefinition();
+      workAddress._instanceMeta.name = 'USPostalAddress';
+      workAddress._instanceMeta.sdType = 'Address';
+      workAddress._instanceMeta.usage = 'Inline';
+      workAddress.country = 'US';
+      workAddress.type = 'postal';
+      workAddress.use = 'work';
+      const homeAddress = new InstanceDefinition();
+      homeAddress._instanceMeta.name = 'USPostalAddress';
+      homeAddress._instanceMeta.sdType = 'Address';
+      homeAddress._instanceMeta.usage = 'Inline';
+      homeAddress.country = 'US';
+      homeAddress.type = 'postal';
+      homeAddress.use = 'home';
+
+      const address = patient.elements.find(e => e.id === 'Patient.address');
+      address.fixValue(workAddress, true);
+      expect(address.fixedAddress).toEqual({ country: 'US', type: 'postal', use: 'work' });
+      address.fixValue(workAddress, true);
+      expect(address.fixedAddress).toEqual({ country: 'US', type: 'postal', use: 'work' });
+      expect(() => {
+        address.fixValue(homeAddress, true);
+      }).toThrow(
+        'Cannot fix {"country":"US","type":"postal","use":"home"} to this element; a different Address is already fixed: {"country":"US","type":"postal","use":"work"}'
+      );
+    });
+
+    it('should throw FixedToPatternError when trying to change fixed[x] to patternp[x]', () => {
```

Comment:
small typo here: `patternp`

Reply:
That pesky "p" key is too close to that pesky "[" key.